dev/tool/src/clean.ts

```diff
@@ -1030,3 +1048,325 @@ export async function restoreHrTaskTypesFromUpdates (
+  try {
+    const workspaces = await listWorkspacesRaw(_client.db(ACCOUNT_DB), productId)
+    workspaces.sort((a, b) => b.lastVisit - a.lastVisit)
+    const initWorkspaces = initWorkspacesStr.split(';')
+    const initWS = workspaces.filter((p) => initWorkspaces.includes(p.workspace))
+    const ids = new Map<string, RelatedDocument[]>()
+    for (const workspace of initWS) {
+      const workspaceId = getWorkspaceId(workspace.workspace, productId)
+      const db = getWorkspaceDB(_client, workspaceId)
+
+      const txex = await db.collection(DOMAIN_TX).find<TxCUD<Doc>>({}).toArray()
+      const txesArr = []
+      for (const obj of txex) {
+        if (obj.objectSpace === core.space.Model && !isPersonAccount(obj)) {
+          continue
+        }
+        txesArr.push({ _id: obj._id, _class: obj._class })
+      }
+      txesArr.filter((it, idx, array) => array.findIndex((pt) => pt._id === it._id) === idx)
+      ids.set(DOMAIN_TX, txesArr)
+
+      const colls = await db.collections()
+      for (const coll of colls) {
+        if (skippedDomains.includes(coll.collectionName)) continue
+        const arr = ids.get(coll.collectionName) ?? []
+        const data = await coll.find<RelatedDocument>({}, { projection: { _id: 1, _class: 1 } }).toArray()
+        for (const obj of data) {
+          arr.push(obj)
+        }
+        ids.set(coll.collectionName, arr)
+      }
+
+      const arr = ids.get(DOMAIN_MODEL) ?? []
+      const data = await db
+        .collection(DOMAIN_TX)
+        .find<TxCUD<Doc>>(
+        { objectSpace: core.space.Model },
+        { projection: { objectId: 1, objectClass: 1, modifiedBy: 1 } }
+      )
+        .toArray()
+      for (const obj of data) {
+        if (
+          (obj.modifiedBy === core.account.ConfigUser || obj.modifiedBy === core.account.System) &&
+          !isPersonAccount(obj)
+        ) {
+          continue
+        }
+        if (obj.objectId === core.account.ConfigUser || obj.objectId === core.account.System) continue
+        arr.push({ _id: obj.objectId, _class: obj.objectClass })
+      }
+      arr.filter((it, idx, array) => array.findIndex((pt) => pt._id === it._id) === idx)
+      ids.set(DOMAIN_MODEL, arr)
+    }
+
+    for (let index = 0; index < workspaces.length; index++) {
+      const workspace = workspaces[index]
+      // we should skip init workspace first time, for case if something went wrong
+      if (initWorkspaces.includes(workspace.workspace)) continue
+
+      ctx.info(`Processing workspace ${workspace.workspaceName ?? workspace.workspace}`)
+      const workspaceId = getWorkspaceId(workspace.workspace, productId)
+      const db = getWorkspaceDB(_client, workspaceId)
+      const check = await db.collection(DOMAIN_MIGRATION).findOne({ state, plugin: workspace.workspace })
+      if (check != null) continue
+
+      const wsClient = (await connect(transactorUrl, workspaceId, undefined, {
```

Comment:
@BykhovDenis maybe we should use local created pipeline instead of transacter connection?

Reply:
I want to disconnect all current user to prevent any problems

Reply:
ok.